modules/lua-loader.nix

```diff
@@ -1,30 +1,32 @@
-{ lib, config, ... }:
-with lib;
+{
+  lib,
+  config,
+  helpers,
+  ...
+}:
 let
   cfg = config.luaLoader;
 in
 {
-  options.luaLoader = {
-    enable = mkOption {
-      type = types.bool;
-      default = false;
-      description = ''
-        Whether to enable/disable the experimental lua loader:
+  options.luaLoader.enable = helpers.mkNullOrOption lib.types.bool ''
+    Whether to enable/disable the experimental lua loader:
 
-        If `true`: Enables the experimental Lua module loader:
-          - overrides loadfile
-          - adds the lua loader using the byte-compilation cache
-          - adds the libs loader
-          - removes the default Neovim loader
+    If `true`: Enables the experimental Lua module loader:
+      - overrides loadfile
+      - adds the lua loader using the byte-compilation cache
+      - adds the libs loader
+      - removes the default Neovim loader
 
-        If `false`: Disables the experimental Lua module loader:
-          - removes the loaders
-          - adds the default Neovim loader
-      '';
-    };
-  };
+    If `false`: Disables the experimental Lua module loader:
+      - removes the loaders
+      - adds the default Neovim loader
+
+    If `null`: Nvim's default is used.
```

Comment:
Do we know what neovim's default is? If so, should we document it like we would for a plugin option?

Reply:
Neovim's default is disabled for now. I considered using library function `defaultNullOpts.mkBool`. It has pluginDefault parameter. But since it's not a _plugin_, I used a plain `mkNullOrOption`. But if plugin default is desirable, I think this option is the way to go.

Reply:
> But since it's not a plugin, I used a plain mkNullOrOption

Agreed. "Plugin Default:" would be confusing.

We could still write "If null: Uses neovim's default (false)" if you want, though that's a maintenance burden if neovim change their default so I don't mind either way.

Reply:
I changed it to "Nothing is configured" which is more accurately reflects the value `null`.